source/BoardingPanel.cpp

```diff
@@ -538,17 +538,34 @@ void BoardingPanel::DoKeyboardNavigation(const SDL_Keycode key)
 BoardingPanel::Plunder::Plunder(const string &commodity, int count, int unitValue)
 	: name(commodity), outfit(nullptr), count(count), unitValue(unitValue)
 {
-	UpdateStrings();
+	UpdateStrings(false);
 }
 
+const std::string BoardingPanel::GenerateOutfitLootName(const PlayerInfo& player,
+														const Outfit *outfit, const int count) const
+{
+	std::string label = outfit->DisplayName();
+
+	if(!player.OutfitIsKnown(*outfit))
+	{
+		label = "Unknown " + outfit->Category();
 
+		if(outfit->Category() == "Power")
+			label += " Component";
+		if((count == 1) && (label.back() == 's'))
+			label.pop_back();
+	}
+
+	return label;
+}
 
 // Constructor (outfit installed in the victim ship or transported as cargo).
-BoardingPanel::Plunder::Plunder(const Outfit *outfit, int count)
-	: name(outfit->DisplayName()), outfit(outfit), count(count),
+BoardingPanel::Plunder::Plunder(const BoardingPanel *boardingPanel, const PlayerInfo &player,
+								const Outfit *outfit, int count)
+	: name(boardingPanel->GenerateOutfitLootName(player, outfit, count)), outfit(outfit), count(count),
```

Comment:
It would seem it would be better to set the correct name in a new function in `Plunder`, instead of one in `BoardingPanel`. So like `UpdateName` maybe?

Reply:
I assume there is a proper way to call a func in Plunder when the object get inited?

Reply:
> It would seem it would be better to set the correct name in a new function in `Plunder`, instead of one in `BoardingPanel`. So like `UpdateName` maybe?

you meant like UpdateStrings? I think you are right, will fix